### PR TITLE
Feature EPL-1482: First version of the Profile Modal

### PR DESCRIPTION
**Pull request by Sankku**

### CODE REVIEW

**Review by tomyrodeghiero — APPROVED** [review]

File: packages/ComponentLibrary/src/components/ProfileModal/ProfileModal.styles.ts [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+  fontWeight: '600',
+  fontSize: '0.875rem',
+  color: 'white',
+  padding: '0.5rem 1rem',
+  background: 'black',
+  flex: '1 0 0',
+  borderRadius: '6.25rem',
+  border: '1px solid rgba(0, 3, 13, 0.20)',
+  height: '2.5rem',
+}
```
Comment (sjcuello) [inline packages/ComponentLibrary/src/components/ProfileModal/ProfileModal.styles.ts:142]: Missing space at the end

File: packages/ComponentLibrary/src/components/ProfileModal/SelectorList/SelectorList.styles.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+export const labelStyles: CSSProperties = {
+  color: PRIMARY_500,
+  fontWeight: '600',
+};
+
+export const iconStyles: CSSProperties = {
+  paddingLeft: '0.5rem'  
+};
+
```
Comment (sjcuello) [inline packages/ComponentLibrary/src/components/ProfileModal/SelectorList/SelectorList.styles.ts:21]: Missing space at the end

File: packages/ComponentLibrary/src/components/ProfileModal/SelectorList/SelectorListMock.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  formStyle,
+} from './SelectorList.styles';
+import RoleIcon from '@mui/icons-material/AccountCircle';
+import ClientIcon from '@mui/icons-material/Business';
+import OrganizationIcon from '@mui/icons-material/Domain';
+import WarehouseIcon from '@mui/icons-material/Store';
+import LanguageIcon from '@mui/icons-material/Language';
+import { SelectorListProps } from './SelectorList.types';
+
+const items = ['Rol', 'Cliente', 'Organización', 'Almacén', 'Lenguaje'];
```
Comment (sjcuello) [inline packages/ComponentLibrary/src/components/ProfileModal/SelectorList/SelectorListMock.tsx:26]: Make an enum in typescript

File: packages/ComponentLibrary/src/components/ProfileModal/SelectorList/SelectorListMock.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+const icons: Record<Item, React.ReactElement> = {
+  Rol: <RoleIcon style={iconStyles} />,
+  Cliente: <ClientIcon style={iconStyles} />,
+  Organización: <OrganizationIcon style={iconStyles} />,
+  Almacén: <WarehouseIcon style={iconStyles} />,
+  Lenguaje: <LanguageIcon style={iconStyles} />,
+};
+
+const SelectorList = <T extends string>({ section }: SelectorListProps<T>) => {
+  const relevantItems = section === 'profile' ? items : [];
```
Comment (sjcuello) [inline packages/ComponentLibrary/src/components/ProfileModal/SelectorList/SelectorListMock.tsx:39]: Compare to a variable in "references".
Rulo may help you with this topic

File: packages/ComponentLibrary/src/components/ProfileModal/SelectorList/SelectorListMock.tsx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+                <div
+                  style={{
+                    display: 'flex',
+                    alignItems: 'center',
+                  }}>
+                  <ListItemIcon>{icons[item]}</ListItemIcon>
+                  <Typography>{`${item} ${selected}`}</Typography>
+                </div>
+              )}>
+              <MenuItem value="1">{item} 1</MenuItem>
```
Comment (sjcuello) [inline packages/ComponentLibrary/src/components/ProfileModal/SelectorList/SelectorListMock.tsx:76]: Be careful here, remember to code as generic as posible

Comment (sjcuello) [reply packages/ComponentLibrary/src/components/ProfileModal/SelectorList/SelectorListMock.tsx:76]: It migth be a param

File: packages/ComponentLibrary/src/components/ProfileModal/UserProfile.tsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  nameStyles,
+  emailStyles,
+  logoutButtonStyles,
+} from './ProfileModal.styles';
+import profilePicture from '../../assets/images/profile_picture.png';
+import BackgroundGradient from '../../assets/images/backgroundGradient.svg';
+import Logout from '@mui/icons-material/Logout';
+
+const UserProfile: React.FC = () => {
+  const user = {
```
Comment (sjcuello) [inline packages/ComponentLibrary/src/components/ProfileModal/UserProfile.tsx:17]: It's a mock.
It doesn't have to be placed here.
Move to a mock file inside this folder

File: packages/ComponentLibrary/src/components/ProfileModal/ProfileModal.tsx [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+const ProfileModal: React.FC = () => {
+  const [currentSection, setCurrentSection] = useState<string>('profile');
+
+  const handleToggle = (selectedSection: string) => {
+    setCurrentSection(selectedSection);
+  };
+
+  return (
+    <Modal width={332}>
```
Comment (leandroallemandismf) [inline packages/ComponentLibrary/src/components/ProfileModal/ProfileModal.tsx:32]: Add to constant

**Review by leandroallemandismf — APPROVED** [review]

**Review by leandroallemandismf — APPROVED** [review]

**Review by leandroallemandismf — APPROVED** [review]

**Pull request merged**